src/Microsoft.ManagedZlib/src/DeflateStream.cs

```diff
@@ -0,0 +1,1078 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.ManagedZLib;
+
+public partial class DeflateStream : Stream
+{
+    private const int DefaultBufferSize = 8192;
+    private Stream _stream;
+    private Inflater? _inflater;
+    private Deflater? _deflater;
+    private byte[]? _buffer;
+    private int _activeAsyncOperation; // 1 == true, 0 == false
```

Comment:
Consider making this a boolean instead, and change the logic that consumes this variable so that it takes a boolean too.